pkg/dashboard/worker/gitops.go

```diff
@@ -388,6 +375,18 @@ func processReleaseEvent(
 			vars[k] = v
 		}
 
+		repoVars, err := loadRepoVars(gitopsRepoCache, event.Repository, ".gimlet/vars")
```

Comment:
There is like 60 lines of code that collect variables from different places.
The 60 lines appear twice: in processReleaseEvent and processArtifactEvent. I feel this could be factored into a function loading all vars.

Also, env vars are read from git on every iteration of `for _, manifest := range artifact.Environments`. This feels slow.